src/gui/C4StartupScenSelDlg.cpp

```diff
@@ -1616,8 +1629,15 @@ void C4StartupScenSelDlg::UpdateList()
 		for (C4ScenarioListLoader::Entry *pEnt = pScenLoader->GetFirstEntry(); pEnt; pEnt = pEnt->GetNext())
 		{
 			if (pEnt->IsHidden()) continue; // no UI entry at all for hidden items
-			ScenListItem *pEntItem = new ScenListItem(pScenSelList, pEnt);
-			if (pEnt == pOldSelection) pScenSelList->SelectEntry(pEntItem, false);
+			if (!SLen(searchBar->GetText()) || SSearchNoCase(pEnt->GetName().getData(), searchBar->GetText()))
```

Comment:
What does the if/else block do here? As far as I understand it: Search bar is clear, or found an item that matches the text => old behaviour. Otherwise => Clear "old selection".

Reply:
Indeed.

Reply:
Could you add a comment that describes that then?